booster/unit-tests/Test/Booster/Pattern/ApplyEquations.hs

```diff
@@ -94,17 +96,17 @@ test_evaluateFunction =
             eval BottomUp subj @?= Right result
         ]
   where
-    eval direction =
-        unsafePerformIO
-            . runNoLoggingT
-            . (fst <$>)
-            . evaluateTerm direction funDef Nothing Nothing mempty
+    eval direction t =
+        unsafePerformIO $ do
+            ns <- noSolver
+            runNoLoggingT $ fst <$> evaluateTerm direction funDef Nothing ns mempty t
```

Comment:
(off-topic)
The `{-# NOINLINE ... #-}` pragmas are certainly a good idea.
However, we can iinstead completely get rid of the `unsafePerformIO`  by making `eval` an `IO` monadic function. The tests that use it run in `IO` anyway! (ditto for the other ones below)